packages/@expo/cli/src/run/resolveRemoteBuildCache.ts

```diff
@@ -0,0 +1,67 @@
+import { ExpoConfig } from '@expo/config';
+import { ModPlatform } from '@expo/config-plugins';
+import spawnAsync from '@expo/spawn-async';
+import fs from 'fs';
+import path from 'path';
+
+const debug = require('debug')('expo:run:remote-build') as typeof console.log;
+
+export async function resolveRemoteBuildCache(
+  projectRoot: string,
+  {
+    platform,
+    provider,
+  }: { platform: ModPlatform; provider?: ExpoConfig['experiments']['remoteBuildCache']['provider'] }
+): Promise<string | null> {
+  const Fingerprint = importFingerprintForDev(projectRoot);
+  if (!Fingerprint) {
+    debug('@expo/fingerprint is not installed in the project, skip checking for remote builds');
+    return null;
+  }
+  const fingerprint = await Fingerprint.createFingerprintAsync(projectRoot);
```

Comment:
this is going to be a bit tricky, since we end up using slightly different defaults with `eas fingerprint:generate` i believe - so we might not get as much re-use from this as we'd like. we can follow up on this afterwards